search-commons/src/main/java/no/unit/nva/search2/model/ResourceSortKeys.java

```diff
@@ -0,0 +1,78 @@
+package no.unit.nva.search2.model;
+
+import static no.unit.nva.search2.constant.ApplicationConstants.UNDERSCORE;
+import static no.unit.nva.search2.constant.Patterns.PATTERN_IS_CASE_INSENSITIVE;
+import static no.unit.nva.search2.constant.Patterns.PATTERN_IS_NONE_OR_ONE;
+import java.util.Arrays;
+import java.util.Collection;
+import java.util.LinkedHashSet;
+import java.util.Locale;
+import java.util.Set;
+import java.util.function.Predicate;
+import java.util.stream.Collectors;
+
+public enum ResourceSortKeys {
+    INVALID(""),
+    CATEGORY("entityDescription.reference.publicationInstance.type"),
+    CONTRIBUTOR_NAME("entityDescription.contributors.identity.name"),
+    CREATED_DATE("createdDate"),
+    INSTITUTION_NAME("entityDescription.contributors.affiliation.name"),
+    MODIFIED_DATE("modifiedDate"),
+    PUBLISHED_DATE("publishedDate"),
+    TITLE("entityDescription.mainTitle"),
+    UNIT_ID("entityDescription.contributors.affiliation.id"),
+    USER("(?i)(user)|(owner)", "resourceOwner.owner");
+
+    public static final Set<ResourceSortKeys> VALID_SORT_PARAMETER_KEYS =
+        Arrays.stream(ResourceSortKeys.values())
+            .sorted(ResourceSortKeys::compareParameterKey)
+            .skip(1)    // skip INVALID
+            .collect(Collectors.toCollection(LinkedHashSet::new));
+
+    private final String pattern;
+    private final String fieldName;
+
+    ResourceSortKeys(String pattern, String fieldName) {
+        this.pattern = pattern;
+        this.fieldName = fieldName;
+    }
+
+    ResourceSortKeys(String fieldName) {
+        var name = this.name().toLowerCase(Locale.getDefault());
+        this.pattern =  PATTERN_IS_CASE_INSENSITIVE + name.replace(UNDERSCORE, PATTERN_IS_NONE_OR_ONE);
```

Comment:
a little hard to understand what these constants do as they are in another class